### PR TITLE
Fix bundle order

### PR DESCRIPTION
**Pull request by acunniffe**
## 🍗 Description
Fixes a bug bundling nested refs

## 📚 References
_Links to relevant docs (Notion, Twist, GH issues, etc.), if applicable._

## 👹 QA
_How can other humans verify that this PR is correct?_

### CODE REVIEW

File: projects/json-pointer-helpers/src/json-pointers/json-pointer-helpers.ts [hunk 1]
```diff
@@ -57,15 +57,21 @@ function relative(pointer: string, from: string) {
   return compile(targetDecoded.slice(fromDecoded.length));
 }
 
-function startsWith(pointer: string, pattern: string[]): boolean {
+function startsWith(
+  pointer: string,
+  pattern: string[],
+  exact: boolean = false
```
Comment (niclim) [inline projects/json-pointer-helpers/src/json-pointers/json-pointer-helpers.ts:63]: Can we change this to be an `options` arg with key exact so it's easier to read when it's used downstream

**Review by niclim — COMMENTED** [review]
Could you explain what this fixes?

**Comment by acunniffe** [comment]
> Could you explain what this fixes?

When we bundle nested references, we need to rewrite JSON Trails ie `/path/method/responses/200/..../schema/.../$ref` becomes `/components/NewName/$ref`. 

The code that decided which references to rewrite was relying on using a string `startsWith` instead of using the JSON Path Matchers. This meant that you might match two similar, but different refs:

```
/...the-same-path/.../customer
/...the-same-path/.../customerContact
```
Would match even though they are siblings and different. The fix uses our existing code to compare the actual path, not their string representation. I added the exact flag as a precaution in case someone had contentType `**/**` which minimatch would respect.

File: projects/json-pointer-helpers/src/json-pointers/json-pointer-helpers.ts (hunk 1 above)
Comment (acunniffe) [reply projects/json-pointer-helpers/src/json-pointers/json-pointer-helpers.ts]: Done

**Review by niclim — APPROVED** [review]
Could be good to include a test here as well

**Pull request merged**